### PR TITLE
feat(rpc)!: add support for Google's new RPC-based request mechanism

### LINKED ISSUE #234: error in result (AttributeError: 'NoneType' object has no attribute 'group')
hi i use Python 3.6.9
and when i call translate have this error:
image

### PR DESCRIPTION
**Pull request by ssut**
:tada: Happy to announce the long-awaited release of such stable Googletrans that responds to the new Google's RPC-based request mechanism. This introduces _possibly_ backward-incompatible API changes and have the following changes:

- `Translator.translate` will no longer return some extra data.
- **confidence** is currently not available.
- The existing methods and classes such as `translate`, `detect` are marked as legacy because they are considered unstable and will be phased out.
- This should be a fairly stable version that fixes #234.

I'm planning to publish this PR as Release Candidate to PyPI starting with [v4.0.0-rc1](https://pypi.org/project/googletrans/4.0.0rc1/). As such use this at your own risk.

```
>>> __import__('googletrans').Translator().translate('안녕하세요. 반가워요!', dest='zh-cn').__dict__()
{'src': 'ko', 'dest': 'zh-cn', 'origin': '안녕하세요. 반가워요!', 'text': '你好。很高兴见到你！', 'pronunciation': 'Nǐ hǎo. Hěn gāoxìng jiàn dào nǐ!', 'extra_data': {'confidence': None, 'parts': [<googletrans.models.TranslatedPart object at 0x106302e80>, <googletrans.models.TranslatedPart object at 0x106302670>], 'origin_pronunciation': 'annyeonghaseyo. bangawoyo!', 'parsed': [['annyeonghaseyo. bangawoyo!', None, 'ko', [[[0, [[[None, 6]], [True]]], [1, [[[None, 7], [7, 12]], [False, True]]]], 12]], [[[None, 'Nǐ hǎo. Hěn gāoxìng jiàn dào nǐ!', None, None, None, [['你好。', ['你好。', '您好。', '你好']], ['很高兴见到你！', ['很高兴见到你！', '很高兴认识你！'], True]]]], 'zh-cn', 1, 'auto'], 'ko']}, 'parts': [{'text': '你好。', 'candidates': ['你好。', '您好。', '你好']}, {'text': '很高兴见到你！', 'candidates': ['很高兴见到你！', '很高兴认识你！']}]}
```
```
>>> __import__('googletrans').Translator().translate('안녕하세요. 반가워요!', dest='en').__dict__()
{'src': 'ko', 'dest': 'en', 'origin': '안녕하세요. 반가워요!', 'text': 'Hello. Nice to meet you!', 'pronunciation': None, 'extra_data': {'confidence': None, 'parts': [<googletrans.models.TranslatedPart object at 0x10632e8b0>, <googletrans.models.TranslatedPart object at 0x10632e0d0>], 'origin_pronunciation': 'annyeonghaseyo. bangawoyo!', 'parsed': [['annyeonghaseyo. bangawoyo!', None, 'ko', [[[0, [[[None, 6]], [True]]], [1, [[[None, 7], [7, 12]], [False, True]]]], 12]], [[[None, None, None, True, None, [['Hello.', ['Hello.', 'Hi.', 'Good morning.']], ['Nice to meet you!', ['Nice to meet you!'], True]]]], 'en', 1, 'auto'], 'ko']}, 'parts': [{'text': 'Hello.', 'candidates': ['Hello.', 'Hi.', 'Good morning.']}, {'text': 'Nice to meet you!', 'candidates': ['Nice to meet you!']}]}
```

### CODE REVIEW

**Comment by jaklinger** [comment]
Thanks for introducing this, I've not had any issues using the new non-bulk `translate` method on OSX10.15.6 or AmazonLinux1/2, python 3.6-3.8. 

It seems however that the bulk operations are more fiddly with the new RPC mechanism. I tried playing around with the raw requests to see whether this would be possible, but with no luck. Have you had any insights into this?

File: googletrans/client.py [hunk 1]
```diff
@@ -46,39 +52,85 @@ class Translator:
...
+            'rpcids': RPC_ID,
+            'bl': 'boq_translate-webserver_20201207.13_p0',
+            'soc-app': 1,
+            'soc-platform': 1,
+            'soc-device': 1,
+            'rt': 'c',
+        }
+        r = self.client.post(url, params=params, data=data)
+
+        if r.status_code != 200 and self.raise_Exception:
```
Comment (AndreaSottana) [inline googletrans/client.py:122]: Should be `self.raise_exception` instead of `self.raise_Exception`

**Comment by ArtanisTheOne** [comment]
Any timetable on when this can be finished or what needs to be done? Also, would detecting lamguages still be done with this version of the module?

**Comment by ArtanisTheOne** [comment]
@ssut

File: googletrans/client.py (hunk 1 above)
Comment (terryyz) [reply googletrans/client.py:122]: @ssut

**Comment by ArtanisTheOne** [comment]
Difference between translate_legacy and translate?

**Pull request closed without merging**